starport/pkg/chaincmd/runner/chain.go

```diff
@@ -181,30 +183,68 @@ func (r Runner) BankSend(ctx context.Context, fromAccount, toAccount, amount str
 		if strings.Contains(err.Error(), "key not found") || // stargate
 			strings.Contains(err.Error(), "unknown address") || // launchpad
 			strings.Contains(b.String(), "item could not be found") { // launchpad
-			return errors.New("account doesn't have any balances")
+			return "", errors.New("account doesn't have any balances")
 		}
 
-		return err
+		return "", err
 	}
 
 	out := struct {
-		Code  int    `json:"code"`
-		Error string `json:"raw_log"`
+		Code   int    `json:"code"`
+		Error  string `json:"raw_log"`
+		TxHash string `json:"txhash"`
 	}{}
 
 	data, err := b.JSONEnsuredBytes()
 	if err != nil {
-		return err
+		return "", err
 	}
 	if err := json.Unmarshal(data, &out); err != nil {
-		return err
+		return "", err
 	}
 
 	if out.Code > 0 {
-		return fmt.Errorf("cannot send tokens (SDK code %d): %s", out.Code, out.Error)
+		return "", fmt.Errorf("cannot send tokens (SDK code %d): %s", out.Code, out.Error)
 	}
 
-	return nil
+	return out.TxHash, nil
+}
+
+// WaitTx waits until a tx is successfully added to a block and can be queried
+func (r Runner) WaitTx(ctx context.Context, txHash string, retryDelay time.Duration, maxRetry int) error {
+	retry := 0
+
+	// retry querying the request
+	checkTx := func() error {
+		stdout := &bytes.Buffer{}
+		if err := r.run(ctx, runOptions{stdout: stdout}, r.chainCmd.QueryTxCommand(txHash)); err != nil {
+			// filter not found error and check for max retry
+			if !strings.Contains(err.Error(), "not found") {
+				return backoff.Permanent(err)
+			}
+			retry++
+			if retry == maxRetry {
+				return backoff.Permanent(fmt.Errorf("can't retrieve tx %s", txHash))
+			}
+			return err
+		}
+
+		// parse tx and check code
+		txResult := struct {
```

Comment:
Can we use JSONEnsuredBytes and use json to unmarshal instead, because we cannot make sure that output will be yaml all the time. https://github.com/tendermint/starport/blob/086e925a7bf6ed271ffea3a1e2d6730da7025747/starport/pkg/chaincmd/runner/chain.go#L129

Reply:
ef5125c